### DISCUSSION TITLE
Selecting VS 2019 compiler for DocFX generator

### DISCUSSION THREAD

**Discussion by pwcrickman** [dotnet/docfx#8394] [Q&A]
I am using an older version of DocFx (2.56.x) with VS 2019.  I have VS 2022 installed on my system.  When I execute DocFx it uses VS 2022 as the inner compiler.  How do I set the DocFx inner compiler to use VS 2019?

**Comment by KalleOlaviNiemitalo** [comment]
From the source code, it appears that DocFX uses MSBuildLocator, which in turn uses hostfxr_resolve_sdk2; so presumably a global.json file could be used to specify the .NET SDK version whose MSBuild you want to use.

**Reply by KalleOlaviNiemitalo** [reply]
Oh, "an older version", so running on .NET Framework… in that case, try running DocFX from Visual Studio Developer Command Prompt of the desired version, or setting the VSINSTALLDIR and VSCMD_VER environment variables manually.

**Reply by pwcrickman** [reply]
I was able to run docfx command in VS Developer Command Prompt.  How do I resolve this error: 

[Failure] Msbuild failed when processing the file '\<csproj path\>' with message: The SDK 'Microsoft.NET.Sdk' specified could not be found.

The csproj and docfx.json file are targeting netcoreapp3.1
